src/util/handlers/BooleanAndHandler.ts

```diff
@@ -0,0 +1,48 @@
+import assert from 'assert';
+import { getLoggerFor } from '../../logging/LogUtil';
+import { InternalServerError } from '../errors/InternalServerError';
+import { promiseEvery } from '../PromiseUtil';
+import { AsyncHandler } from './AsyncHandler';
+import { filterHandlers } from './HandlerUtil';
+
+/**
+ * A composite handler that returns true if all of its handlers can handle the input and return true.
+ * Handler errors are interpreted as false results.
+ */
+export class BooleanAndHandler<TIn> extends AsyncHandler<TIn, boolean> {
+  protected readonly logger = getLoggerFor(this);
+
+  private readonly handlers: AsyncHandler<TIn, boolean>[];
+
+  /**
+   * Creates a new BooleanAndHandler that stores the given handlers.
+   * @param handlers - Handlers over which it will run.
+   */
+  public constructor(handlers: AsyncHandler<TIn, boolean>[]) {
+    super();
+    this.handlers = handlers;
+  }
+
+  public async canHandle(input: TIn): Promise<void> {
+    // We use this to generate an error if some handler does not support the input
+    assert(
+      (await filterHandlers(this.handlers, input)).length === this.handlers.length,
+      'Not all handlers support this input.',
+    );
+  }
```

Comment:
What is the reason from doing this one differently from the `BooleanOrHandler`? It can still handle the input, it's just going to return `false` as a result. Besides that the classes are quite similar so perhaps this could be a single class that takes a boolean as constructor parameter to know whether all or some should return true? Similar to the `UnionHandler`.

Reply:
Makes sense.